apps/dgst.c

```diff
@@ -403,7 +403,7 @@ int dgst_main(int argc, char **argv)
     if (md == NULL) {
         EVP_MD_CTX *tctx;
         BIO_get_md_ctx(bmd, &tctx);
-        md = EVP_MD_CTX_md(tctx);
+        md = (EVP_MD *)EVP_MD_CTX_md(tctx);
```

Comment:
This change, or rather the `EVP_MD_free()` of this pointer down at the end, causes the following test error, which can now be observed in the CIs

```
free(): double free detected in tcache 2
../../util/wrap.pl ../../apps/openssl dgst -sign ../../test/testrsa.pem -out testrsa.sig ../../test/data.bin => 134
```